src/version/version.rs

```diff
@@ -44,42 +45,55 @@ fn get_installed_version() -> Result<Version, failure::Error> {
     Ok(parsed_version)
 }
 
-fn check_wrangler_versions() -> Result<WranglerVersion, failure::Error> {
-    let current = get_installed_version()?;
-    let latest = get_latest_version(&current.to_string())?;
-    Ok(WranglerVersion { current, latest })
-}
-
-fn get_latest_version(installed_version: &str) -> Result<Version, failure::Error> {
+fn check_wrangler_versions() -> Result<(WranglerVersion, bool), failure::Error> {
     let config_dir = get_wrangler_home_dir()?;
     let version_file = config_dir.join("version.toml");
     let current_time = SystemTime::now();
-    let latest_version = match fs::read_to_string(&version_file) {
-        Ok(contents) => {
-            let last_checked_version = LastCheckedVersion::from_str(&contents)?;
+
+    let mut checked = false;
+    let current = get_installed_version()?;
+
+    let latest = match get_version_disk(&version_file)? {
+        Some(last_checked_version) => {
             let time_since_last_checked =
                 current_time.duration_since(last_checked_version.last_checked)?;
             if time_since_last_checked.as_secs() < ONE_DAY {
-                let version = Version::parse(&last_checked_version.latest_version)?;
-                Some(version)
-            } else {
-                None
+                checked = true;
             }
+            Version::parse(&last_checked_version.latest_version)?
         }
-        Err(_) => None,
+        // If version.toml doesn't exist, fetch latest version
+        None => get_latest_version(&current.to_string(), &version_file, current_time)?,
     };
-    match latest_version {
-        Some(latest_version) => Ok(latest_version),
-        None => {
-            let latest_version = get_latest_version_from_api(installed_version)?;
-            let updated_file_contents = toml::to_string(&LastCheckedVersion {
-                latest_version: latest_version.to_string(),
-                last_checked: current_time,
-            })?;
-            fs::write(&version_file, updated_file_contents)?;
-            Ok(latest_version)
+
+    Ok((WranglerVersion { current, latest }, checked))
```

Comment:
this tuple return is awkward. will review in the morning and suggest a different tactic.